src/main/java/org/stroganov/ws/handlers/UserValidator.java

```diff
@@ -0,0 +1,104 @@
+package org.stroganov.ws.handlers;
+
+import jakarta.xml.soap.*;
+import jakarta.xml.ws.handler.MessageContext;
+import jakarta.xml.ws.handler.soap.SOAPHandler;
+import jakarta.xml.ws.handler.soap.SOAPMessageContext;
+import jakarta.xml.ws.soap.SOAPFaultException;
+import org.apache.log4j.Logger;
+import org.stroganov.dao.LibraryDAO;
+import org.stroganov.entities.User;
+import org.stroganov.util.DataManager;
+
+import javax.xml.namespace.QName;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Set;
+
+public class UserValidator implements SOAPHandler<SOAPMessageContext> {
+    public static final String CREDENTIALS_ERROR = "Не удалось получить правильный пароль для пользователя:";
+    private final Logger logger = Logger.getLogger(UserValidator.class);
+    protected final LibraryDAO libraryDAO = DataManager.getLibraryDAO();
+    protected User user = null;
```

Comment:
Not the best idea to store the user in a field, there may be problems with multithreading. If the same handler object will be used by different threads.
```suggestion
    
```

Reply:
Good notice! I fixed it.